### PR TITLE
producer chart: job/cronjob switch

### PR DESCRIPTION
**Pull request by mkcode92**
* make producer-app deployable as job
* problem: when switching between job/cronjob a lot of shared config is needed in a different path (spec.template vs spec.jobTemplate.spec.template)
* this block is now separated as podTemplate

### CODE REVIEW

File: charts/producer-app/templates/job.yaml [hunk 1]
```diff
@@ -11,86 +16,16 @@ metadata:
...
+  suspend: {{ .Values.suspend }}
   {{- if .Values.schedule }}
   schedule: {{ quote .Values.schedule }}
-  {{- end }}
-  suspend: {{ .Values.suspend }}
   successfulJobsHistoryLimit: 1
   concurrencyPolicy: Replace
   jobTemplate:
     spec:
       template:
```
Comment (yannick-roeder) [inline charts/producer-app/templates/job.yaml:26]: Did you test this in the cluster? This is going to be an empty key when no schedule is set

**Comment by yannick-roeder** [comment]
LGTM, did you test it in the cluster?

**Comment by torbsto** [comment]
@VictorKuenstler  @philipp94831 This is a fix for the changes introduced by #108. I've mistakenly assumed that Job and CronJob have the same spec besides `schedule`. Obviously, the changes in this PR are more complex than those in #108. I still think it's worth to have it.

@mkcode92 I think you also have to update the `producer-app` chart according to the steps listed here: https://github.com/bakdata/streams-bootstrap/tree/master/charts#development

**Comment by yannick-roeder** [comment]
> @mkcode92 I think you also have to update the `producer-app` chart according to the steps listed here: https://github.com/bakdata/streams-bootstrap/tree/master/charts#development

True, thanks @torbsto

**Comment by mkcode92** [comment]
> LGTM, did you test it?

Yes, both should work!

**Review by yannick-roeder — APPROVED** [review]

**Review by yannick-roeder — APPROVED** [review]

**Pull request merged**